csv_page_result.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"strconv"
+	"strings"
+)
+
+type csvPageResult struct {
+	URL   string
+	Links []csvLinkResult
+}
+
+type csvLinkResult struct {
+	URL    string
+	Status string
+}
+
+func newCSVPageResult(r *pageResult, verbose bool) *csvPageResult {
+	c := len(r.ErrorLinkResults)
+
+	if verbose {
+		c += len(r.SuccessLinkResults)
+	}
+
+	ls := make([]csvLinkResult, 0, c)
+
+	if verbose {
+		for _, r := range r.SuccessLinkResults {
+			ls = append(ls, csvLinkResult{
+				URL:    r.URL,
+				Status: strconv.Itoa(r.StatusCode),
+			})
+		}
+	}
+
+	for _, r := range r.ErrorLinkResults {
+		ls = append(ls, csvLinkResult{
+			URL:    r.URL,
+			Status: r.Error.Error(),
+		})
+	}
+
+	return &csvPageResult{r.URL, ls}
+}
+
+func (r *csvPageResult) String() string {
+	var buf strings.Builder
+
+	buf.WriteString(`"Page URL","Link URL",Status` + "\n")
+
+	for _, link := range r.Links {
+		buf.WriteString(`"` + strings.ReplaceAll(r.URL, `"`, `""`) + `","` +
```

Comment:
Can we use the `encoding/csv` package?

Reply:
I wanted to be accurate and ensure that string fields are always quoted, since URL's technically can have comma's (as per RFC 3986). For example in the query string. My first attempt was actually using the `encode/csv` package, however read somewhere that it doesn't quote by default or otherwise quote every field, so opted for this approach. 

However, after some more reading, I think my understanding may have been wrong and can actually use the `encoding/csv` instead. I'll do some refactoring :)